docs/syntax.rst

```diff
@@ -236,44 +305,63 @@ ForwardRef_
 
   # stenotype
 
-----
-
-
 Special Function Qualifiers
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-You can use a limited set of keywords to describe a number of special functions
-in ``stenotype`` when annotating a function object as a variable.
+You can use a limited set of keywords to describe a number of common types.
+These are especially useful when annotating the return types of functions.
+Keywords correspond to how the type is used to get a value of a specifi type.
 
 **Asynchronous function**
 
 .. code-block:: python
 
+  # value
+  async def foo(a: A, b: B) -> R:
+    return r
+
+  bar = foo(a, b)
+
   # regular
-  typing.Callable[[A, B], typing.Awaitable[R]
+  foo: typing.Callable[[A, B], typing.Awaitable[R]]
+  bar: typing.Awaitable[R]
 
   # stenotype
-  "(A, B) -> await R"
+  foo: "(A, B) -> await R"
+  bar: "await R"
 
 **Asynchronous generator**
 
 .. code-block:: python
 
+  # value
+  async def foo(a: A, b: B) -> AsyncIterable[R]:
+     yield r
+  
+  bar = foo(a, b)
+
   # regular
-  typing.Callable[[A, B], typing.AsyncIterable[R]
+  foo: typing.Callable[[A, B], typing.AsyncIterable[R]]
+  bar: typing.AsyncIterable[R]
 
   # stenotype
-  "(A, B) -> await *R"
+  foo: "(A, B) -> await *R"
+  bar: "await *R"
 
 **Context managing functions**
 
 .. code-block:: python
 
+  @contextmanager
+  def foo(a: A, b: B):
+      yield r
```

Comment:
this should return, right?

Reply:
It's a [``contextlib.contextmanager``](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager) which does need the ``yield`` -- do you think that is too obscure? I don't know if there are any well-known context managers in the standard library.